### DISCUSSION TITLE
How can we see the data in test_data.input?

### DISCUSSION THREAD

**Discussion by aichicaideyang** [awslabs/gluonts#3159] [Q&A]
I have some questions about the Example in the readme. It seems that the last three years of data are used for prediction. However, when splitting the test_gen, we don't know what context_length is. So, when predicting for the year 1958, should the input data be for the year 1957? Is there any way to see what the data looks like in test_data.input? The encapsulated class is too complex, and I can't see what it contains specifically.

**Comment by c1adrien** [comment]
Hey, `context_length` refers to the amount of historical data you will pass to your neural network for making predictions. For example, in the case of a DeepVAR model (during training), you pass a tensor of size `(batch_size, context_length + prediction_length, ...)` to your LSTM unit.

If you want to inspect what's inside a `dataset.train` object (which is an iterable of DataRow), you can use:

```next(iter(dataset.train))```
Hope this helps!
